fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java

```diff
@@ -2845,8 +2845,8 @@ private String getRequestBaseUri(String type) throws Exception {
             }
         }
 
-        // Strip any path segments for whole-system interactions (in case of whole-system search, "Resource" is passed as the type)
-        if (type == null || type.isEmpty() || "Resource".equals(type)) {
+        // Strip any path segments for whole-system interactions (in case of whole-system search, "Resource" is passed as the type, or $everything-based search)
+        if (type == null || type.isEmpty() || "Resource".equals(type) || baseUri.contains("$everything")) {
```

Comment:
is this being trimmed for a reason?

Reply:
Yeah, otherwise the resulting `fullUrl` of the patient resources resulting from the search will have *$everything* in there too.